gemrb/GUIScripts/LUCommon.py

```diff
@@ -59,6 +59,28 @@ def CanLevelUp(actor):
 		next = GUIREC.GetNextLevelExp(levelsum, GUIREC.GetECL(actor))
 		return next <= xp
 
+	# hardcoded special case to handle TNO, who is usually a single class 
+	# but with three separate Levels/XP values and the ability to switch between them
+	# it returns the active class if true
+	Switcher = GUICommon.IsNamelessOne(actor) 
+	if Switcher:
+		xp = [ GemRB.GetPlayerStat (actor, IE_XP), GemRB.GetPlayerStat (actor, IE_XP_MAGE), GemRB.GetPlayerStat (actor, IE_XP_THIEF) ]
```

Comment:
You can add a dict to map the class names to the xp/level index, use it to get the correct index regardless of results and then avoid the code triplication below. Less chance of mistakes that way too — now you have too many Levels[0] checks.